### PR TITLE
Worked on the language of the Quasiquote documentation

### PR DESCRIPTION
**Pull request by derekwyatt**
- There wasn't anything "wrong" with the documentation
- This is really just a set of changes that improve the grammar so it's
  a bit more readable

### CODE REVIEW

File: overviews/quasiquotes/definition-details.md [hunk 1]
```diff
@@ -12,13 +12,13 @@ outof: 13
 
 ## Modifiers
 
-Every definition except packages and package objects have associated modifiers object which contains following data:
+Every definition except packages and package objects have associated an modifiers object that contains the following data:
```
Comment (SethTisue) [inline overviews/quasiquotes/definition-details.md]: s/an modifiers/a modifiers/

Comment (SethTisue) [inline overviews/quasiquotes/definition-details.md]: s/an modifiers/a modifiers/

File: overviews/quasiquotes/definition-details.md [hunk 2]
```diff
@@ -58,32 +58,32 @@ In deconstruction one can either extract `Modifiers` or annotations, but you can
...
+1. Early definitions. A list of `val` or `type` definitions. Type definitions are still allowed by they are deprecated and will be removed in the future:
 
         scala> val withx = q"new { val x = 1 } with RequiresX"
         withx: universe.Tree = ...
 
         scala> val q"new { ..$earlydefns } with RequiresX" = withx
         earlydefns: List[universe.Tree] = List(val x = 1)
 
-2. List of parents. A list of type identifiers with possibly an optional arguments to the first one in the list:
+2. List of parents. A list of type identifiers with optional arguments to the first one in the list:
```
Comment (SethTisue) [inline overviews/quasiquotes/definition-details.md]: this sentence remains unclear (but if you don't know how to fix it, feel free to just leave it)

File: overviews/quasiquotes/definition-details.md [hunk 3]
```diff
@@ -126,15 +126,15 @@ Each val and var consistents of four components: modifiers, name, type tree and
     tpt: universe.Tree = <type ?>
     rhs: universe.Tree = 2
 
-If type of the val isn't explicitly specified by the user an [empty type](/overviews/quasiquotes/type-details.html#empty-type) is used as tpt.
+If the type of the `val` isn't explicitly specified by the user an [empty type](/overviews/quasiquotes/type-details.html#empty-type) is used as `tpt`.
 
-Vals and vars are disjoint (they don't match one another):
+`Val`s and `var`s are disjoint (they don't match one another):
```
Comment (SethTisue) [inline overviews/quasiquotes/definition-details.md]: If it's in backquotes, it's literal and shouldn't have an initial capital letter, even at the start of a sentence.

File: overviews/quasiquotes/expression-details.md [hunk 4]
```diff
@@ -62,9 +62,9 @@ Similarly it would work with all the literal types except `Null`. (see [standard
 
 ## Identifier and Selection
 
-Identifiers and member selections are two fundamental primitives that let you refer to other definitions. Combination of two of them is also known `RefTree`.
+Identifiers and member selections are two fundamental primitives that let you refer to other definitions. A combination of two of them is also known `RefTree`.
```
Comment (SethTisue) [inline overviews/quasiquotes/expression-details.md]: also known as a

File: overviews/quasiquotes/expression-details.md [hunk 5]
```diff
@@ -75,20 +75,20 @@ Each term identifier is defined by its name and by the fact of being backquoted
     scala> val backquoted = q"`$name`"
     backquoted: universe.Ident = `Foo`
 
-Although backquoted and non-backquoted identifiers may refer to the same things they are not equivalent from syntactical point of view:
+Although backquoted and non-backquoted identifiers may refer to the same thing they are not equivalent from a syntactical point of view:
```
Comment (SethTisue) [inline overviews/quasiquotes/expression-details.md]: Although backquoted and non-backquoted identifiers may refer to the same thing, they are not syntactically equivalent:

File: overviews/quasiquotes/expression-details.md [hunk 6]
```diff
@@ -75,20 +75,20 @@ Each term identifier is defined by its name and by the fact of being backquoted
...
+This is because backquoted identifiers have different semantics in pattern patching.
 
-Apart from matching on identifiers with given name you can also extract their name values with the help of [unlifting](/overviews/quasiquotes/unlifting.html):
+Apart from matching on identifiers with a given name, you can also extract their name values with the help of [unlifting](/overviews/quasiquotes/unlifting.html):
 
     scala> val q"${name: TermName}" = q"Foo"
     name: universe.TermName = Foo
 
-Name ascription is important here as without it you\'ll get pattern that is equivalent to regular pattern variable binding.
+Name ascription is important here because without it you\'ll get a pattern that is equivalent to regular pattern variable binding.
```
Comment (SethTisue) [inline overviews/quasiquotes/expression-details.md]: throughout the document, I don't think putting backslash before single quote is necessary

File: overviews/quasiquotes/expression-details.md [hunk 7]
```diff
@@ -156,11 +156,11 @@ It's recommended to always include type applications when you match on function
     ts: List[universe.Tree] = List(Int)
     args: List[universe.Tree] = List(1, 2, 3)
 
-Another important feature of scala method calls is multiple argument lists and implicit arguments:
+Other important features of scala method calls are multiple argument lists and implicit arguments:
```
Comment (SethTisue) [inline overviews/quasiquotes/expression-details.md]: s/scala/Scala/

**Comment by SethTisue** [comment]
@densh want to give this a once-over before merge?

File: overviews/quasiquotes/definition-details.md (hunk 2 above)
Comment (derekwyatt) [inline overviews/quasiquotes/definition-details.md]: I made the others, but you're on your own with this one :)

Comment (densh) [inline overviews/quasiquotes/definition-details.md]: What it means that in q"new foo with bar", you can put type and value arguments next foo, but not next to bar. This is caused by the fact that the first parent has to be a class, while others are just traits (which don't accept value arguments yet.)

File: overviews/quasiquotes/expression-details.md [hunk 8]
```diff
@@ -565,7 +557,7 @@ is critical:
 
 ## For and For-Yield Loops
 
-For and For-Yield expressions allow to write monadic style comprehensions that desugar into calls to `map`, `flatMap`, `foreach` and `withFilter` methods:
+For and For-Yield expressions allow us to write a monadic style comprehension that desugar into calls to `map`, `flatMap`, `foreach` and `withFilter` methods:
```
Comment (densh) [inline overviews/quasiquotes/expression-details.md]: For and For-Yield should probably be backquoted too for consistency with val and var.

File: overviews/quasiquotes/hygiene.md [hunk 9]
```diff
@@ -10,11 +10,11 @@ outof: 13
...
 
-Sophisticated macro systems such as Racket's have mechanisms that make macros hygienic without any effort from macro writers. In Scala we don't have automatic hygiene yet - both of our codegen facilities (compile-time codegen with macros and runtime codegen with toolboxes) require programmers to handle hygiene manually. Fixing this is our number one priority for 2.12 (see [future prospects](/overviews/quasiquotes/future.html)), but in the meanwhile you need to know how to work around the absence of hygiene, which is what this section is about.
+Sophisticated macro systems such as Racket's have mechanisms that make macros hygienic without any effort from macro writers. In Scala we don't have automatic hygiene yet - both of our codegen facilities (compile-time codegen with macros and runtime codegen with toolboxes) require programmers to handle hygiene manually. Fixing this is our number one priority for 2.12 (see [future prospects](/overviews/quasiquotes/future.html)) but, in the meantime, you must know how to work around the absence of hygiene, which is what this section is about.
```
Comment (densh) [inline overviews/quasiquotes/hygiene.md]: I'd remove `yet`, nobody is working on this any longer.

Comment (densh) [inline overviews/quasiquotes/hygiene.md]: The `Fixing this is our number one priority for 2.12...` sentence is out of date too and should be removed altogether.

Comment (derekwyatt) [inline overviews/quasiquotes/hygiene.md]: (sad panda)

**Pull request merged**